pr_watch/tests/test_models.py

```diff
@@ -0,0 +1,164 @@
+        self.assertEqual(watched_pr.updates_feed, 'https://github.com/open-craft/edx/commits/test-branch.atom')
+
+    def test_set_fork_name(self):
+        """
+        Set org & repo using the fork name
+        """
+        watched_pr = WatchedPullRequest()
+        watched_pr.set_fork_name('org2/another-repo')
+        self.assertEqual(watched_pr.github_organization_name, 'org2')
+        self.assertEqual(watched_pr.github_repository_name, 'another-repo')
+        watched_pr.save()
+
+        # Check values in DB
+        watched_pr = WatchedPullRequest.objects.get(pk=watched_pr.pk)
+        self.assertEqual(watched_pr.github_organization_name, 'org2')
+        self.assertEqual(watched_pr.github_repository_name, 'another-repo')
+
+    def test_get_branch_tip(self):
+        """
+        Set the commit id to the tip of the current branch, using the default commit policy (True)
+        """
+        self.mock_get_commit_id_from_ref.return_value = 'b' * 40
+        watched_pr = WatchedPullRequest(
+            github_organization_name='org3',
+            github_repository_name='repo3',
+        )
+        self.assertEqual(watched_pr.get_branch_tip(), 'b' * 40)
+        self.assertEqual(self.mock_get_commit_id_from_ref.mock_calls, [
+            call('org3/repo3', 'master', ref_type='heads'),
+        ])
+
+    def test_get_branch_tip_with_tag(self):
+        """
+        Set the commit id to a tag.
+
+        TODO: Is this 'ref_type' code used for anything?
+        """
+        self.mock_get_commit_id_from_ref.return_value = 'c' * 40
+        instance = WatchedPullRequest.objects.create(
+            fork_name='org9/repo',
+            branch_name='new-tag',
+            ref_type='tag',
+        )
+        self.assertEqual(instance.get_branch_tip(), 'c' * 40)
+        self.assertEqual(self.mock_get_commit_id_from_ref.mock_calls, [
+            call('org9/repo', 'new-tag', ref_type='tag'),
+        ])
+        self.assertEqual(instance.branch_name, 'new-tag')
+        self.assertEqual(instance.ref_type, 'tag')
+
+    @override_settings(INSTANCE_EPHEMERAL_DATABASES=True)
+    def test_create_from_pr(self):
+        """
+        Create an instance from a pull request
+        """
+        pr = PRFactory()
+        instance, created = WatchedPullRequest.objects.update_or_create_from_pr(pr)
+        self.assertTrue(created)
+
+        watched_pr = instance.watchedpullrequest
+        self.assertEqual(watched_pr.instance, instance)
+        self.assertEqual(watched_pr.github_pr_number, pr.number)
+        self.assertEqual(watched_pr.fork_name, pr.fork_name)
+        self.assertEqual(watched_pr.branch_name, pr.branch_name)
+
+        self.assertEqual(instance.sub_domain, 'pr{}.sandbox'.format(pr.number))
+        self.assertRegex(instance.name, r'^PR')
+        self.assertEqual(instance.edx_platform_commit, '9' * 40)
+        self.assertTrue(instance.use_ephemeral_databases)
+
+    @override_settings(INSTANCE_EPHEMERAL_DATABASES=True)
```

Comment:
@bradenmacdonald Why do we need to override settings here? If a PR description specifies info about using ephemeral databases, shouldn't it override settings automatically? (Genuine question - applies to next test as well.)